daemon.c

```diff
@@ -1540,8 +1556,10 @@ void *receiveOOK(void *param) {
 					/* Let's do a little filtering here as well */
 					if(r.length >= hw->minrawlen && r.length <= hw->maxrawlen) {
 						receive_queue(r.pulses, r.length, plslen, hw->hwtype);
+						r.length = 0;
 					}
-					r.length = 0;
+					// Avoid that short footer pulses (X10, IMPULS, ...) interfere with long header /sync pulses
+					if(r.length>20)r.length = 0;	// Avoid a buffer reset and preserve potential Header / SYNC / Start pulses
```

Comment:
As said in the comments:

The debug code that was detecting and reporting the fact (that a protocol manipulates the crucial system parameters hw->mingaplen, hw->maxgaplen) got removed, but we still we do need to pay attention to it, as we need to continue to collect data for the first 20 pulses of a s*mfy pulsetrain even if the duration of those pulses are higher than hw->mingaplen.

X10 and IMPULS protocols interfere with s*mfy protocol, as they set hw->mingaplen to a value that is lower than the duration of data pulses of the s*mfy protocol.

 

Von: CurlyMoo [mailto:notifications@github.com] 
Gesendet: Sonntag, 20. Dezember 2015 14:31
An: pilight/pilight
Cc: wo-rasp
Betreff: Re: [pilight] S*mfy Protocol (#200)

 

In daemon.c <https://github.com/pilight/pilight/pull/200#discussion_r48103398> :

>                                     }
> -                                   r.length = 0;
> +                                   // Avoid that short footer pulses (X10, IMPULS, ...) interfere with long header /sync pulses
> +                                   if(r.length>20)r.length = 0;   // Avoid a buffer reset and preserve potential Header / SYNC / Start pulses

What's up with this line?

—
Reply to this email directly or view it on GitHub <https://github.com/pilight/pilight/pull/200/files#r48103398> .  <https://github.com/notifications/beacon/AG9MV0kWcDkPikBvqEo91gBdd_bMeOwPks5pRqUjgaJpZM4GUJeH.gif>